File: repositories/conta_repository.py

from models.conta_saldo import ContaSaldo
from interfaces.i_conta_repository import IContaRepository

class ContaRepository(IContaRepository):
    def __init__(self):
        self.tabela_saldos = [
            ContaSaldo(938485762, 180),
            ContaSaldo(347586970, 1200),
            ContaSaldo(2147483649, 0),
            ContaSaldo(675869708, 4900),
            ContaSaldo(238596054, 478),
            ContaSaldo(573659065, 787),
            ContaSaldo(210385733, 10),
            ContaSaldo(674038564, 400),
            ContaSaldo(563856300, 1200)
        ]

    def get_saldo(self, conta_id):
        return next((x for x in self.tabela_saldos if x.conta_id == conta_id), None)

    def atualizar_saldo(self, conta_saldo):
        try:
            for i, conta in enumerate(self.tabela_saldos):
                if conta.conta_id == conta_saldo.conta_id:
                    self.tabela_saldos[i] = conta_saldo
                    return True
            
            self.tabela_saldos.append(conta_saldo)
            return True
        except Exception as error:
            print(f"Erro ao atualizar saldo: {str(error)}")
            return False

File: repositories/ContaRepository.js

const ContaSaldo = require('../models/ContaSaldo');

class ContaRepository {
    constructor() {
        this.tabelaSaldos = [
            new ContaSaldo(938485762, 180),
            new ContaSaldo(347586970, 1200),
            new ContaSaldo(2147483649, 0),
            new ContaSaldo(675869708, 4900),
            new ContaSaldo(238596054, 478),
            new ContaSaldo(573659065, 787),
            new ContaSaldo(210385733, 10),
            new ContaSaldo(674038564, 400),
            new ContaSaldo(563856300, 1200)
        ];
    }

    getSaldo(contaId) {
        return this.tabelaSaldos.find(x => x.contaId === contaId);
    }

    atualizarSaldo(contaSaldo) {
        try {
            const index = this.tabelaSaldos.findIndex(x => x.contaId === contaSaldo.contaId);
            if (index !== -1) {
                this.tabelaSaldos[index] = contaSaldo;
            } else {
                this.tabelaSaldos.push(contaSaldo);
            }
            return true;
        } catch (error) {
            console.log(`Erro ao atualizar saldo: ${error.message}`);
            return false;
        }
    }
}

module.exports = ContaRepository;
